### PR TITLE
Study+Service metadata

### PR DESCRIPTION
**Pull request by ignapas**
## What do these changes do?

* Removed intermediate container in NodePorts

* Changed layout in service settings.
  * Added service title in a top toolbar.
  * Placed buttons inside the toolbar, instead of at the bottom.
  * Improved display of iframe in fullscreen mode.
  * Inputs
    * Added a fixed title bar
    * Improved layout structure
    * Added a collapse all button
    * Inputs now share the entire vertical space

* Resolved some bug in the panel view

* Setting hints
  * When focusing on a setting an info icon shows up. When hovering, a tooltip containing a hint about the setting is shown

* Replaces Overview panel for a more useful Study information panel, showing the most relevant study metadata to the user.

## Screenshots
before
Screenshot 2019-06-19 at 11 34 42
after
Screenshot 2019-06-19 at 11 27 35
before
Screenshot 2019-06-19 at 11 37 07
after
Screenshot 2019-06-19 at 11 39 56
Screenshot 2019-06-25 at 16 49 13
Screenshot 2019-07-02 at 14 04 53
hints

## Related issue number

#854 #859 #855

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js [hunk 1]
```diff
@@ -130,11 +135,17 @@ qx.Class.define("qxapp.component.workbench.NodeUI", {
...
-      this.add(this.__progressBar);
+      this.add(this.__createChipContainer());
+
+      if (this.getNode().getMetaData().category === "Solver") {
+        this.__progressBar = new qx.ui.indicator.ProgressBar().set({
+          height: 10,
+          margin: 4
+        });
+        this.add(this.__progressBar);
+      } else if (this.getNode().getMetaData().type === "dynamic") {
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: this.getNode().isDynamic()

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js [hunk 2]
```diff
@@ -130,11 +135,17 @@ qx.Class.define("qxapp.component.workbench.NodeUI", {
...
       });
 
-      this.__progressBar = new qx.ui.indicator.ProgressBar().set({
-        height: 10,
-        margin: 4
-      });
-      this.add(this.__progressBar);
+      this.add(this.__createChipContainer());
+
+      if (this.getNode().getMetaData().category === "Solver") {
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: this.getNode().isComputational() ?

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js [hunk 3]
```diff
@@ -150,7 +161,9 @@ qx.Class.define("qxapp.component.workbench.NodeUI", {
         this.__createUIPorts(true, metaData.inputs);
         this.__createUIPorts(false, metaData.outputs);
       }
-      node.bind("progress", this.__progressBar, "value");
+      if (node.getMetaData().category === "Solver") {
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: this.getNode().isComputational() ?

File: services/web/client/source/class/qxapp/statics/Node.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/*
+ * oSPARC - The SIMCORE frontend - https://osparc.io
+ * Copyright: 2019 IT'IS Foundation - https://itis.swiss
+ * License: MIT - https://opensource.org/licenses/MIT
+ * Authors: Ignacio Pascual (ignapas)
+ */
+
+qx.Class.define("qxapp.statics.Node", {
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/statics/Node.js]: Coud you, please, rename it to NodeIcon or something else?

**Comment by odeimaiz** [comment]
Could you also, please, when a Node or Study is a member of a class make it property? (qxapp.component.service.NodeStatus and qxapp.component.metadata.StudyInfo)

**Comment by odeimaiz** [comment]
As I mentioned some time ago, drag&dropping from the inputs to the settings might be a long way. Other than that (I didn't test it) code and UI look very nice.

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js (hunk 2 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: This is only for solvers... The file picker is also set as computational (maybe we change that in the future), but only solvers will actually show a progress bar

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js (hunk 1 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: Done

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js (hunk 3 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: Only solvers get a progress bar

File: services/web/client/source/class/qxapp/statics/Node.js (hunk 4 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/statics/Node.js]: Done

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js (hunk 2 above)
Comment (odeimaiz) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: As far as I know, we don't have Computational services other than Solvers. Am I missing something?

Comment (ignapas) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: Alright

**Comment by ignapas** [comment]
> Appearance looks great.
> The only think I would consider is to follow gitlab's philosophy issue labels for the services,
> 
> i.e. have a panel of labels like [this](https://github.com/ITISFoundation/osparc-simcore/labels)
> 
> image
> 
> and render them on top of the services as this
> image
> 
> Then the labels can be used to filter (upon click on them) and in the search entry.

@pcrespov It looks like a really nice new and useful functionality. It will come in a new PR

**Comment by ignapas** [comment]
> Could you also, please, when a Node or Study is a member of a class make it property? (qxapp.component.service.NodeStatus and qxapp.component.metadata.StudyInfo)

I try to always make them properties when they need to be accessed or changed at a later point from the outside. In this case they are only used internally for rendering these components' GUIs. The thing is that if you add them as properties, then you have to handle the possible future ``setPropertyName`` by implementing an ``applyPropertyName`` function that updates the layout, and this will never occur in these cases.

File: services/web/client/source/class/qxapp/component/workbench/NodeUI.js (hunk 3 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/workbench/NodeUI.js]: Done

**Pull request merged**